test/test_logging_interface.cpp

```diff
@@ -130,16 +130,20 @@ class LoggingTest : public ::testing::Test
 
   void TearDown()
   {
-    if (std::filesystem::remove_all(test_log_dir_) == 0) {
-      std::cerr << "Failed to remove " << test_log_dir_.string() << " directory\n";
+    if (std::filesystem::exists(test_log_dir_) && std::filesystem::is_directory(test_log_dir_)) {
+      if (std::filesystem::remove_all(test_log_dir_) == 0) {
```

Comment:
Note that we cannot do the following. so we check the directory exists 1st before removing.

```bash
/root/ros2_ws/colcon_ws/src/ros2/rcl_logging_syslog/test/test_logging_interface.cpp: In member function ‘virtual void LoggingTest::TearDown()’:
/root/ros2_ws/colcon_ws/src/ros2/rcl_logging_syslog/test/test_logging_interface.cpp:133:52: warning: comparison of unsigned expression in ‘< 0’ is always false [-Wtype-limits]
  133 |     if (std::filesystem::remove_all(test_log_dir_) < 0) {
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~
/root/ros2_ws/colcon_ws/src/ros2/rcl_logging_syslog/test/test_logging_interface.cpp: In member function ‘bool LoggingTest::remove_test_folder()’:
/root/ros2_ws/colcon_ws/src/ros2/rcl_logging_syslog/test/test_logging_interface.cpp:140:52: warning: comparison of unsigned expression in ‘< 0’ is always false [-Wtype-limits]
  140 |     if (std::filesystem::remove_all(test_log_dir_) < 0) {
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~
```